src/agent/transform-headers-agent.ts

```diff
@@ -36,12 +37,19 @@ class TransformHeadersAgent extends WrappedAgent {
             }
         }
 
+        const typedRequest = request as ClientRequest & {
+            _removedContLen: boolean;
+            _contentLength: number;
+            _removedTE: boolean;
+            agent?: Agent;
+        };
+
         if (!hasConnection) {
-            const shouldSendKeepAlive = request.shouldKeepAlive && (hasContentLength || request.useChunkedEncodingByDefault || request.agent);
+            const shouldSendKeepAlive = request.shouldKeepAlive && (hasContentLength || request.useChunkedEncodingByDefault || typedRequest.agent);
             if (shouldSendKeepAlive) {
-                headers.Connection = 'keep-alive';
+                headers['Connection'] = 'keep-alive'; // eslint-disable-line dot-notation
             } else {
-                headers.Connection = 'close';
+                headers['Connection'] = 'close'; // eslint-disable-line dot-notation
```

Comment:
Is this a conflict between ESLint and TS expectations?

Reply:
Correct